src/formats/internal/memreader.c

```diff
@@ -83,22 +96,38 @@ int8_t memread_byte(memreader *reader) {
 
 int16_t memread_word(memreader *reader) {
     int16_t r;
+#ifdef BIG_ENDIAN_BUILD
+    r = __builtin_bswap16(*((uint16_t *)(reader->buf + reader->pos)));
+#else
     memcpy(&r, reader->buf + reader->pos, sizeof(r));
+#endif
     reader->pos += sizeof(r);
     return r;
 }
 
 int32_t memread_dword(memreader *reader) {
     int32_t r;
+#ifdef BIG_ENDIAN_BUILD
+    r = __builtin_bswap32(*((int32_t *)(reader->buf + reader->pos)));
+#else
     memcpy(&r, reader->buf + reader->pos, sizeof(r));
+#endif
     reader->pos += sizeof(r);
     return r;
 }
 
 float memread_float(memreader *reader) {
     float r;
+#ifdef BIG_ENDIAN_BUILD
+    uint32_t fl;
+    fl = *((uint32_t *)(reader->buf + reader->pos));
```

Comment:
Please use a memcpy, as the buf+pos pointer is insufficiently aligned to be a `(uint32_t *)` .
```diff
-    fl = *((uint32_t *)(reader->buf + reader->pos));
+    memcpy(&fl, reader->buf + reader->pos, sizeof(fl));
```